File: VICSApp/CoreSystem/DHCD/FormUI/Mainform.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Configuration
Imports System.Data.SqlClient
Imports System.Threading
Imports System.Globalization
Imports BenlyDAL.BenlyDAL
Public Class Mainform
    Public conn As New SqlConnection
    Public BenlyDal As New DAL(conn)
    Public workingmeeting As String = ""
    Public dateMeeting As DateTime = DateTime.Now


    Private Sub Mainform_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Thread.CurrentThread.CurrentCulture = New CultureInfo("vi-VN")
    End Sub

    Private Sub Mainform_Shown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shown
        connect2db()
        MeetingList.Show()
    End Sub
    Private Sub connect2db()
        If conn.State = ConnectionState.Closed Then
            conn.ConnectionString = GetConnectionString("connDBstr")
            Try
                conn.Open()
            Catch ex As Exception
                Throw ex
                Exit Sub
            End Try
        End If
    End Sub
    Public Function GetConnectionString(ByVal key As String) As String
        Dim strApp As String = ""
        Try
            strApp = ConfigurationManager.ConnectionStrings(key).ConnectionString()
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        Return strApp
    End Function
    Public Function addthousandseperator(ByVal str As String) As String
        Dim result As String = str
        Dim len As Integer = str.Length
        If len > 3 Then
            result = result.Insert(len - 3, ".")
            len = len - 3

            While len > 3
                result = result.Insert(len - 3, ".")
                len = len - 3
            End While

        End If
        Return result
    End Function
    Private Sub DanhSáchCuộcHọpToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchCuộcHọpToolStripMenuItem.Click
        MeetingList.Show()
    End Sub

    Private Sub DanhSáchCổĐôngToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchCổĐôngToolStripMenuItem.Click
        HolderList.Show()
    End Sub

    Private Sub DanhSáchĐạiBiểuToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchĐạiBiểuToolStripMenuItem.Click
        DelegateList.Show()
    End Sub

    Private Sub DanhSáchỦyQuyềnToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchỦyQuyềnToolStripMenuItem.Click
        AuthorizationList.Show()
    End Sub

    Private Sub DanhSáchVấnĐềBiểuQuyếtToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchVấnĐềBiểuQuyếtToolStripMenuItem.Click
        MatterList.Show()
    End Sub

    Private Sub BiểuQuyếtToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BiểuQuyếtToolStripMenuItem.Click

    End Sub

    Private Sub DanhSáchVấnĐềBầuCửToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchVấnĐềBầuCửToolStripMenuItem.Click
        ElectionList.Show()
    End Sub

    Private Sub DanhSáchỨngViênToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchỨngViênToolStripMenuItem.Click
        CandidateList.Show()
    End Sub

    Private Sub ToolStripMenuItem1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripMenuItem1.Click
        MatterVoteList.Show()
    End Sub

    Private Sub KếtQuảBầuCửToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles KếtQuảBầuCửToolStripMenuItem.Click
        Elections_result.Show()
    End Sub

    Private Sub DanhSáchPhiếuBầuCửToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchPhiếuBầuCửToolStripMenuItem.Click
        ElectionVoteList.Show()
    End Sub

    Private Sub KếtQuảBiểuQuyếtToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles KếtQuảBiểuQuyếtToolStripMenuItem.Click
        MatterVoteResult.Show()
    End Sub

    Private Sub ThôngTinCuộcHọpToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ThôngTinCuộcHọpToolStripMenuItem.Click
        MeetingInfoshow.ShowDialog()
    End Sub

    Private Sub DanhSáchPhiếuBầuKhôngHợpLệToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DanhSáchPhiếuBầuKhôngHợpLệToolStripMenuItem.Click
        IllegalElectionList.Show()
    End Sub

    Private Sub AbcToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        MessageBox.Show("abc")
    End Sub
End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Mainform.cs
using System;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Globalization;
using System.Threading;
using System.Windows.Forms;
using BenlyDAL.BenlyDAL;
using Microsoft.VisualBasic;

namespace pmDHCD
{
    public partial class Mainform
    {
        public Mainform()
        {
            BenlyDal = new DAL(conn);
            InitializeComponent();
            _DanhSáchCuộcHọpToolStripMenuItem.Name = "DanhSáchCuộcHọpToolStripMenuItem";
            _DanhSáchCổĐôngToolStripMenuItem.Name = "DanhSáchCổĐôngToolStripMenuItem";
            _DanhSáchĐạiBiểuToolStripMenuItem.Name = "DanhSáchĐạiBiểuToolStripMenuItem";
            _DanhSáchỦyQuyềnToolStripMenuItem.Name = "DanhSáchỦyQuyềnToolStripMenuItem";
            _ThôngTinCuộcHọpToolStripMenuItem.Name = "ThôngTinCuộcHọpToolStripMenuItem";
            _BiểuQuyếtToolStripMenuItem.Name = "BiểuQuyếtToolStripMenuItem";
            _DanhSáchVấnĐềBiểuQuyếtToolStripMenuItem.Name = "DanhSáchVấnĐềBiểuQuyếtToolStripMenuItem";
            _ToolStripMenuItem1.Name = "ToolStripMenuItem1";
            _KếtQuảBiểuQuyếtToolStripMenuItem.Name = "KếtQuảBiểuQuyếtToolStripMenuItem";
            _DanhSáchVấnĐềBầuCửToolStripMenuItem.Name = "DanhSáchVấnĐềBầuCửToolStripMenuItem";
            _DanhSáchỨngViênToolStripMenuItem.Name = "DanhSáchỨngViênToolStripMenuItem";
            _DanhSáchPhiếuBầuCửToolStripMenuItem.Name = "DanhSáchPhiếuBầuCửToolStripMenuItem";
            _KếtQuảBầuCửToolStripMenuItem.Name = "KếtQuảBầuCửToolStripMenuItem";
            _DanhSáchPhiếuBầuKhôngHợpLệToolStripMenuItem.Name = "DanhSáchPhiếuBầuKhôngHợpLệToolStripMenuItem";
        }

        public SqlConnection conn = new SqlConnection();
        public DAL BenlyDal;
        public string workingmeeting = "";
        public string period = "";
        public string mettingType = "";
        public string year = DateTime.Today.Year.ToString();
        public DateTime dateMeeting = DateTime.Now;

        private void Mainform_Load(object sender, EventArgs e)
        {
            Thread.CurrentThread.CurrentCulture = new CultureInfo("vi-VN");
        }

        private void Mainform_Shown(object sender, EventArgs e)
        {
            connect2db();
            My.MyProject.Forms.MeetingList.Show();
        }

        private void connect2db()
        {
            if (conn.State == ConnectionState.Closed)
            {
                conn.ConnectionString = GetConnectionString("connDBstr");
                try
                {
                    conn.Open();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi kết nối DB: "+ex.Message);
                    return;
                }
            }
        }

        public string GetConnectionString(string key)
        {
            string strApp = "";
            try
            {
                strApp = ConfigurationManager.ConnectionStrings[key].ConnectionString;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }

            return strApp;
        }

        public string addthousandseperator(string str)
        {
            string result = str;
            int len = str.Length;
            if (len > 3)
            {
                result = result.Insert(len - 3, ".");
                len = len - 3;
                while (len > 3)
                {
                    result = result.Insert(len - 3, ".");
                    len = len - 3;
                }
            }

            return result;
        }

        private void DanhSáchCuộcHọpToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.MeetingList.Show();
        }

        private void DanhSáchCổĐôngToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.HolderList.Show();
        }

        private void DanhSáchĐạiBiểuToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.DelegateList.Show();
        }

        private void DanhSáchỦyQuyềnToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.AuthorizationList.Show();
        }

        private void DanhSáchVấnĐềBiểuQuyếtToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.MatterList.Show();
        }

        private void BiểuQuyếtToolStripMenuItem_Click(object sender, EventArgs e)
        {
        }

        private void DanhSáchVấnĐềBầuCửToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.ElectionList.Show();
        }

        private void DanhSáchỨngViênToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.CandidateList.Show();
        }

        private void ToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.MatterVoteList.Show();
        }

        private void KếtQuảBầuCửToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Elections_result.Show();
        }

        private void DanhSáchPhiếuBầuCửToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.ElectionVoteList.Show();
        }

        private void KếtQuảBiểuQuyếtToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.MatterVoteResult.Show();
        }

        private void ThôngTinCuộcHọpToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.MeetingInfoshow.ShowDialog();
        }

        private void DanhSáchPhiếuBầuKhôngHợpLệToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.IllegalElectionList.Show();
        }

        private void AbcToolStripMenuItem_Click(object sender, EventArgs e)
        {
            MessageBox.Show("abc");
        }
    }
}